### ISSUE TITLE
Eclipse error when using generics with nested Builder

### ISSUE THREAD

**Issue by mcliedtke** [immutables/immutables#633]
I am having trouble with immutables in eclipse. I am using eclipse 4.6.3 and the Immutables 2.5.2 jar for annotation processing.

The problem is illustrated in the follow case:

```java
public abstract class A {}
```
```java
public class B extends A {}
```
```java
public class C extends A {}
```
```java
@Value.Immutable
public abstract class D<T extends A> {
    
    public abstract T someField();
    
    public static class Builder<T extends A> extends ImmutableD.Builder<T> {}
}
```

This ends up compiling fine via `javac` but, for whatever reason, the `ImmutableD` class is never generated while inside eclipse.

Interestingly enough, if I change the nested builder line to the following: 

```java
...
    public static class Builder<T> extends ImmutableD.Builder<T> {}
...
```
The `ImmutableD` class gets generated by eclipse, but Immutables throws an error saying on that same line declaring that "Inner type Builder should have the same type parameters as abstract value type:  \<T extends some.test.package.A\>"

I guess I have two questions:
1. Have you seen this before or are you able to replicate the problem in eclipse?
2. Does the nested `Builder` really need to have the declaration `<T extends A>`, isn't `<T>` sufficient?

Last note is that if I change the Type parameter on `D` to `public abstract class D<T>`, then everything works as expected, but it is not the level of typing that I am after.

**Comment by elucash** [comment]
1. It's clearly a reproducible bug
2. If declare just `T`, the builder would be unable to produce instance of `D<T>` because `T` has to have a type compatible with the `extends A` bound
The fix will be released shortly. Thank you for spotting this!

**Comment by mcliedtke** [comment]
Thank you Eugene, I really appreciate the quick turnaround! Absolutely love Immutables, keep up the good work! 👍

**Comment by elucash** [comment]
released in 2.5.3 thank you!
